module_utils/bsize.py

```diff
@@ -0,0 +1,151 @@
+
+    def _parse_units(self, raw_units):
+        '''
+            gets string containing size units and
+            returns *_FACTOR (BINARY or DECIMAL) and the prefix position (not index!)
+            in the PREFIXES_* list
+            If no unit is specified defaults to BINARY and Bytes
+        '''
+
+        prefix = raw_units
+        # get rid of possible units suffix ('bytes', 'b' or 'B')
+        for suffix in SUFFIXES:
+            if raw_units.lower().endswith(suffix.lower()):
+                prefix = raw_units[:-len(suffix)]
+                break
+
+        if prefix == "":
+            # no unit was specified, use default
+            return BINARY_FACTOR, 0
+
+        # check the list for units
+        idx = -1
+
+        for lst in PREFIXES_DECIMAL:
+            lower_lst = [x.lower() for x in lst]
+            if prefix.lower() in lower_lst:
+                idx = lower_lst.index(prefix.lower())
+                used_factor = DECIMAL_FACTOR
+                break
+
+        if idx < 0:
+            for lst in PREFIXES_BINARY:
+                lower_lst = [x.lower() for x in lst]
+                if prefix.lower() in lower_lst:
+                    idx = lower_lst.index(prefix.lower())
+                    used_factor = BINARY_FACTOR
+                    break
+
+        if idx < 0:
+            raise ValueError("Unable to identify unit '%s'" % raw_units)
+
+        return used_factor, idx+1
+
+    def _parse_number(self, raw_number):
+        ''' parse input string containing number
+            return float
+        '''
+        return float(raw_number)
+
+    def _get_unit(self, factor, exponent, unit_type=0):
+        ''' based on decimal or binary factor and exponent
+            obtain and return correct unit
+        '''
+
+        if unit_type == 0:
+            suffix = "B"
+        else:
+            suffix = "bytes"
+
+        if exponent == 0:
+            return suffix
+
+        if factor == DECIMAL_FACTOR:
+            prefix_lst = PREFIXES_DECIMAL[unit_type]
+        else:
+            prefix_lst = PREFIXES_BINARY[unit_type]
+        return prefix_lst[exponent-1] + suffix
+
+    @property
+    def bytes(self):
+        ''' returns size value in bytes as int
+        '''
+        return int((self.factor ** self.exponent) * self.number)
+
+    def _format(self, format_str, factor, exponent):
+
+        result = format_str
+        result = result.replace(r"%sb", self._get_unit(factor, exponent, 0))
+        result = result.replace(r"%lb", self._get_unit(factor, exponent, 1))
+
+        return result
+
+    def get(self, units='autobin', fmt="%0.1f %sb"):
+        ''' returns size value as a string with given units and format
+
+            "units" parameter allows to select preferred unit.
+                accepted values are also:
+                'autobin' (default) - uses the highest human readable unit (binary)
+                'autodec' - uses the highest human readable unit (decimal)
```

Comment:
Since there are only two choices here I would suggest a boolean flag 'decimal' that defaults to `None` or `False`. It eliminates the free-form string and simplifies the method signature. I'd also make that flag the last argument.

Reply:
Hmm, maybe name it something else since there is a `decimal` module in the standard library of python.